t/t2601-job-shell-standalone.t

```diff
@@ -0,0 +1,155 @@
+	grep -q task_count=1 echo.err
+'
+test_expect_success 'flux-shell: missing JOBID fails with Usage message' '
+	test_must_fail ${FLUX_SHELL} -s -r 0 -j j1 -R R1 2>nojobid.err &&
+	grep -q Usage nojobid.err
+'
+test_expect_success 'flux-shell: missing -r fails with standalone message' '
+	test_must_fail ${FLUX_SHELL} -s -j j1 -R R1 0 2>no_r.err &&
+	grep -q standalone no_r.err
+'
+test_expect_success 'flux-shell: missing -R fails with standalone message' '
+	test_must_fail ${FLUX_SHELL} -s -r 0 -j j1 0 2>no_R.err &&
+	grep -q standalone no_R.err
+'
+test_expect_success 'flux-shell: missing -j fails with standalone message' '
+	test_must_fail ${FLUX_SHELL} -s -r 0 -R R1 0 2>no_j.err &&
+	grep -q standalone no_j.err
+'
+test_expect_success 'flux-shell: nonexistent jobspec file fails' '
+	! ${FLUX_SHELL} -v -s -r 0 -j /noexist -R R1 0 \
+		>noexist.out 2>noexist.err &&
+	grep -q "error opening" noexist.err
+'
+test_expect_success 'flux-shell: malformed jobid fails' '
+	! ${FLUX_SHELL} -v -s -r 0 -j j1 -R R1 BADID \
+		>badid.out 2>badid.err &&
+	grep -q "jobid" badid.err
+'
+test_expect_success 'flux-shell: out of range jobid fails' '
+	! ${FLUX_SHELL} -v -s -r 0 -j j1 -R R1 18446744073709551616 \
+		>badid2.out 2>badid2.err &&
+	grep -q "jobid" badid2.err
+'
+test_expect_success 'flux-shell: out of range broker rank fails' '
+	test_must_fail ${FLUX_SHELL} -s -r 4294967296 -j j1 -R R1 0 2>er.err &&
+	grep -q -i option er.err
+'
+test_expect_success 'flux-shell: wrong range broker rank fails' '
+	test_must_fail ${FLUX_SHELL} -s -r 8 -j j1 -R R1 0 2>wr.err &&
+	grep -q -i fetching wr.err
+'
+test_expect_success 'flux-shell: unknown argument fails' '
+	test_must_fail ${FLUX_SHELL} --FOO -s -r 0 -j j1 -R R1 0 2>alien.err &&
+	grep -q -i unrecognized alien.err
+'
+test_expect_success 'flux-shell: generate 2-task jobspec and matching R' '
+	flux jobspec srun -N1 -n2 printenv >j2 &&
+	cat >R2 <<-EOT
+	{"version": 1, "execution":{ "R_lite":[
+		{ "children": { "core": "0-1" }, "rank": "0" }
+        ]}}
+	EOT
+'
+test_expect_success 'flux-shell: run 2-task printenv job' '
+	${FLUX_SHELL} -v -s -r 0 -j j2 -R R2 42 \
+		>printenv.out 2>printenv.err
+'
+test_expect_success 'flux-shell: 0: completed with no error' '
+	grep -q "task 0 complete status=0" printenv.err
+'
+test_expect_success 'flux-shell: 1: completed with no error' '
+	grep -q "task 1 complete status=0" printenv.err
+'
+test_expect_success 'flux-shell: 0: FLUX_TASK_LOCAL_ID, TASK_RANK set' '
+	grep -q FLUX_TASK_LOCAL_ID=0 printenv.out &&
+	grep -q FLUX_TASK_RANK=0 printenv.out
+'
+test_expect_success 'flux-shell: 1: FLUX_TASK_LOCAL_ID, TASK_RANK set' '
+	grep -q FLUX_TASK_LOCAL_ID=1 printenv.out &&
+	grep -q FLUX_TASK_RANK=1 printenv.out
+'
+test_expect_success 'flux-shell: FLUX_JOB_SIZE, JOB_NNODES, JOB_ID set' '
+	grep -q FLUX_JOB_SIZE=2 printenv.out &&
+	grep -q FLUX_JOB_NNODES=1 printenv.out &&
+	grep -q FLUX_JOB_ID=42 printenv.out &&
+	! grep -q FLUX_URI printenv.out
```

Comment:
The test for !`FLUX_URI` should be split into a different test here, IMO -- since the test could fail even though `FLUX_JOB_SIZE` `JOB_NNODES` and `JOB_ID` are all set (which could confuse the viewer)

Also, I noticed that this test fails when the standalone test is run under an existing instance because of this code in `shell_task_create`:
```C
    flux_cmd_unsetenv (task->cmd, "FLUX_URI");
    if (getenv ("FLUX_URI")) {
        if (flux_cmd_setenvf (task->cmd, 1, "FLUX_URI", "%s",
                              getenv ("FLUX_URI")) < 0)
            goto error;
    }
```

If a `FLUX_URI` is already set in the user's environment (which is the case when running under an instance or probably in future on a system with a Flux system instance), then even in standalone mode the shell will propagate `FLUX_URI`.

To fix, you could force unset FLUX_URI during the test.

Reply:
Additionally, I'd suggest removal of all `-q` arguments to `grep` in these tests. sharness already suppresses stdout when tests are run normally, and the `-q` makes it harder to debug the tests when run with `-d -v`...